internal/codegen/transform_test.go

```diff
@@ -0,0 +1,56 @@
+package codegen
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestToConfigCase(t *testing.T) {
+	testCases := []struct {
+		input    string
+		expected string
+	}{
+		{"GitHub", "GitHub"},
+		{"HMACAlgorithms", "HmacAlgorithms"},
+		{"Shopify", "Shopify"},
+		{"TokenIO", "TokenIo"},
+		{"AWS SNS", "Awssns"},
+		{"3dEye", "3DEye"},
+		{"NMI Payment Gateway", "NmiPaymentGateway"},
+		{"VerificationHMACConfigs", "VerificationHmacConfigs"},
```

Comment:
This is one of the biggest issue with the codegen that I ran into. The naming convention is a bit confusing from the SDK. For example, with something like "aws_sns" / "AWS SNS", from the SDK, sometimes we use `AwsSns`, sometimes we use `Awssns`, so it's a bit confusing here.

From there, "HMACAlgorithms" -> "HmacAlgorithms" and "TokenIO" -> "TokenIo" are also difficult to make sense of.

Here are the functions we use from the AWS SNS example:

```go
func (p *awsSnsSourceVerificationProvider) ToPayload(sourceVerification *SourceVerification) *hookdeck.VerificationConfig {
	if sourceVerification.AwsSns == nil {
		return nil
	}
	return hookdeck.NewVerificationConfigFromAwsSns(&hookdeck.VerificationAwssns{
		Configs: &hookdeck.VerificationAwssnsConfigs{},
	})
}
```

As you can see, the casing is very awkward.

I ended up having to hard-code some "exceptions" because there's no clear rules here. If possible, a conversation with the Fern team would be helpful to improve the codegen script here.